src/odemis/dataio/csv.py

```diff
@@ -45,159 +45,260 @@ def export(filename, data):
-            unit = "nm"
-
-        # attach distance as first row
-        line_length = data.shape[0] * data.metadata[model.MD_PIXEL_SIZE][1]
-        distance_lin = numpy.linspace(0, line_length, data.shape[0])
-        distance_lin.shape = (distance_lin.shape[0], 1)
-        data = numpy.append(distance_lin, data, axis=1)
-
-        # Data should be in the form of (X, C+1), with the first row and column the
-        # distance_from_origin\wavelength
-        with open(filename, 'w', newline='') as fd:
-            csv_writer = csv.writer(fd)
-            # Set the 'header' in the 0,0 element
-            first_row = ['distance_from_origin(m)\\wavelength(' + unit + ')'] + spectrum_range
-            csv_writer.writerow(first_row)
-            # dump the array
-            csv_writer.writerows(data)
-
-    elif data.ndim == 2 and dims == "TC":
+    # Export Temporal Spectrum Data
+    elif dims == "TC" and data.ndim == 2:
         logging.debug("Exporting temporal spectrum data to CSV")
-        if acq_type != model.MD_AT_TEMPSPECTRUM:
-            logging.warning("Data seems to be temporal spectrum, but acq_type is %s", acq_type)
-
-        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
-        if unit_c == "m":
-            spectrum_range = [s * 1e9 for s in spectrum_range]
-            unit_c = "nm"
-        time_range, unit_t = spectrum.get_time_range(data)
-        if unit_t == "s":
-            time_range = [t * 1e12 for t in time_range]
-            unit_t = "ps"
-
-        headers = ["time(" + unit_t + ")\\wavelength(" + unit_c + ")"] + spectrum_range
-        rows = [(t,) + tuple(d) for t, d in zip(time_range, data)]
-
-        with open(filename, 'w', newline='') as fd:
-            csv_writer = csv.writer(fd)
-            csv_writer.writerow(headers)
-            csv_writer.writerows(rows)
-
-    elif data.ndim == 2 and dims == "AC":
-        logging.debug("Exporting angular spectrum data to CSV")
-        if acq_type != model.MD_AT_EK:
-            logging.warning("Data seems to be EK, but acq_type is %s", acq_type)
+        _export_temporal_spectrum_data(acq_type, data, filename)
 
-        spectrum_range, unit_c = spectrum.get_spectrum_range(data)
-        if unit_c == "m":
-            spectrum_range = [s * 1e9 for s in spectrum_range]
-            unit_c = "nm"
+    # Export Angular Spectrum Data
+    elif dims == "AC" and data.ndim == 2:
+        logging.debug("Exporting angular spectrum data to CSV")
+        _export_angular_spectrum_data(acq_type, data, filename)
 
-        angle_range, unit_a = spectrum.get_angle_range(data)
-        if unit_a == "rad":
-            unit_a = "°"
-            angle_range = [math.degrees(theta) for theta in angle_range]  # Convert radians to degrees
-            assert len(angle_range) == len(data)
+    # Export AR Data
+    elif acq_type == model.MD_AT_AR:
+        logging.debug("Exporting AR data to CSV")
+        _export_ar_data(data, filename)
```

Comment:
You moved this to the end of the `elif` statements, with `elif` order does matter. I'm not sure if it matters in this case, but I would move this back to in between "Angle-line data" and "Spectrum data"

Reply:
Yes you are right, elif order does matter but since this specific elif checks completely different parameters I thought putting it on a different spot would make it more stand out. Does this check not have dimensions to check for as parameters? For now I moved it back to its original position as I do not want the code to behave different.